Firestore/core/src/firebase/firestore/util/suppress_warnings.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_UTIL_SUPPRESS_WARNINGS_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_UTIL_SUPPRESS_WARNINGS_H_
+
+/**
+ * Macros for suppressing various warnings. Use like this:
+ *
+ * SUPPRESS_DOCUMENTATION_WARNINGS_BEGIN()
+ * #include "header/that/has/warnings.h"
+ * SUPPRESS_END()
+ *
+ * SUPPRESS_DEPRECATED_DECLARATIONS_BEGIN()
+ * #include "header/that/uses/deprecated/stuff.h"
+ * SUPPRESS_END()
+ *
+ * SUPPRESS_DOCUMENTATION_WARNINGS_BEGIN()
+ * SUPPRESS_DEPRECATED_DECLARATIONS_BEGIN()
+ * #include "something/awful.h"
+ * SUPPRESS_END()
+ * SUPPRESS_END()
+ */
+
+// Define some primitives used by the 'public' macros below.
+
+#if defined(__clang__) || defined(__GNUC__)
+#define SUPPRESS_BEGIN_UNIMPLEMENTED_() _Pragma("GCC diagnostic push")
```

Comment:
I may have given you bad advice. This is failing the Mac cmake build. Maybe the clang version really needs to be `clang diagnostic push`?

Reply:
No, your advice was good. I just can't spell. Twice. (Line 43: disagnostic? Also line 55: GNU vs GNUC.)